### ISSUE TITLE
CLI support

### ISSUE THREAD

**Issue by flabbergastedbd** [owtf/reboot#3]
It is a necessity to support command line invocation let it testing or any other reason. So, instead of writing CLI support code directly, I recommend building a full fledged API and then translating CLI input to use that API. Any thoughts are welcome!

**Comment by DePierre** [comment]
tl;dr: One concern, efficiency. Faster to use internal functions than send HTTP request to REST API methods.

I wouldn't put testing first but scripting and quick execution.
I know that the current CLI in OWTF is a real mess. I am not talking about whether it is working or not (FYI it is not, it is 90% broken). I am talking about how the CLI is integrated in OWTF.

The problem is that the CLI is in owtf.py, the entry point of OWTF but some CLI options need to access the configuration, meaning the CLI has to wait for the full framework to load sometimes. Also, even after processing the CLI, some OWTF components need to access some CLI information but they can't because the CLI was only available in owtf.py, meaning that we had to hack around to keep some copies of some parameters somewhere for some components.

What could be done for the reboot is the following:
- Have one dedicated component for the CLI options (could be a merge of the CLI and the Options component in the current OWTF version for instance)

Concerning the "translating CLI input to use the API" I have one concern: efficiency. Between calling a function in python and sending a HTTP request to a REST API method, I already know which one is faster, even locally!
Instead, since each exposed API method will rely on an internal function, better translate the CLI input to use theses functions instead.

@tunnelshade Does it make sense?

**Comment by flabbergastedbd** [comment]
@DePierre Yes, efficiency takes a hit if we support the CLI using API. So, we will go with a dedicated component which will translate the CLI into python function calls which handle the API.

**Comment by marioskourtesis** [comment]
I like the idea of having APIs but I am not sure if this can be a
replacement of the CLI.

**Comment by DePierre** [comment]
A RESTful API cannot replace a CLI per se imo. If you have nice examples of such cases, that could change my mind though.
